dnf/rpm/transaction.py

```diff
@@ -61,6 +63,18 @@ def dbMatch(self, *args, **kwds):
             mi.pattern(tag, tp, pat)
         return mi
 
+    def dbCookie(self):
+        # dbCookie() does not support lazy opening of rpm database.
+        # The following line opens the database if it is not already open.
+        self.ts.openDB()
+
+        cookie = self.ts.dbCookie()
+        if not cookie:
+            _logger.error(_('The dbCookie() function did not return cookie of rpm database.'))
```

Comment:
What about to also add: `RPM DB can be corrupted`

Reply:
`RPM DB can be corrupted` this is one of the reason. E.g. hash algorithm is not accessible is another - it is the problem with SHA1 and FIPS we just solving.

Reply:
I added a test to check that the database was successfully opened.
An unopened database is another reason why a cookie will not be returned.

Reply:
@lukash 
> Does librpm not report any details about the error that occurred?

Some errors are logged to the rpmlog. E.g. `rpmtsOpenDB` function implementation contains the line:
`rpmlog(RPMLOG_ERR, _("cannot open Packages database in %s\n"), dn);`

Reply:
Well logging is nice and everything but it's not really a replacement for correct error handling.

Reply:
@jrohel Thank you for clarification, Then please keep it like it is.

Reply:
We don't know the reason why the rpm database opening failed. Or why the cookie was not returned. We only know, it fails. And librpm logs message in some situations. We have not more details from librm.

Reply:
I blame librpm :see_no_evil: